test/tictactoeTest.js

```diff
@@ -0,0 +1,831 @@
+                }
+              },
+              "jsonapi": {
+                "version": "1.0"
+              }
+            }
+          )
+        );
+
+        clearButton.click();
+
+        saveButton.click();
+
+        expect(requests[1].method).to.equal('POST');
+        expect(requests[1].url).to.equal('/games');
+      });
+    });
+  });
+
+  describe('Completing a game', () => {
+    beforeEach(() => {
+      xhr = sinon.useFakeXMLHttpRequest();
+      requests = [];
+      xhr.onCreate = req => requests.push(req);
+    });
+
+    afterEach(() => {
+      xhr.restore();
+      resetFixtures();
+    });
+
+    it('auto-saves tie games', () => {
+      squares[0].innerHTML = squares[2].innerHTML = squares[3].innerHTML = squares[5].innerHTML = 'X';
+      squares[1].innerHTML = squares[4].innerHTML = squares[6].innerHTML = squares[8].innerHTML = 'O';
+      //  X | O | X 
+      // -----------
+      //  X | O | X 
+      // -----------
+      //  O |   | O 
+
+      window.turn = 8;
+      window.doTurn(squares[7]);
+
+      expect(requests[0].method).to.equal('POST');
+      expect(requests[0].url).to.equal('/games');
+    });
+
+    it('auto-saves won games', () => {
+      squares[0].innerHTML = squares[1].innerHTML = 'X';
+      squares[6].innerHTML = squares[7].innerHTML = 'O';
+      window.turn = 4;
+      //  X | X |   
+      // -----------
+      //    |   |   
+      // -----------
+      //  O | O |   
+
+      squares[2].click();
+
+      expect(requests[0].method).to.equal('POST');
+      expect(requests[0].url).to.equal('/games');
+    });
+  });
+
+  describe('Clicking a saved game button (in the div#games element)', () => {
+    beforeEach(() => {
+      xhr = sinon.useFakeXMLHttpRequest();
+      requests = [];
+      xhr.onCreate = req => requests.push(req);
+    });
+
+    afterEach(() => {
+      xhr.restore();
+      resetFixtures();
+    });
+
+    it('sends a GET request to the "/games/:id" route', () => {
+      previousButton.click();
+
+      requests[0].respond(
+        200,
+        { 'Content-Type': 'application/json' },
+        JSON.stringify(
+          {
+            "data": [
+              {
+                "id": "1",
+                "type": "games",
+                "attributes": {
+                  "state": ["", "", "", "", "X", "", "", "O", ""]
+                }
+              }
+            ],
+            "jsonapi": {
+              "version": "1.0"
+            }
+          }
+        )
+      );
+
+      const gameButtons = Array.from(gamesDiv.children).filter(c => c.tagName === 'BUTTON');
+
+      gameButtons[0].click();
+
+      expect(requests[1].method).to.equal('GET');
+      expect(requests[1].url).to.equal('/games/1');
+    });
+
+    it("loads the saved game's state into the board", () => {
+      previousButton.click();
+
+      requests[0].respond(
+        200,
+        { 'Content-Type': 'application/json' },
+        JSON.stringify(
+          {
+            "data": [
```

Comment:
There must be a way to write a quick function to abstract this out. You've repeated it at least three times